hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -1025,4 +1028,41 @@ private static void checkNotPagingPredicate(Predicate predicate, String method)
             throw new IllegalArgumentException("PagingPredicate not supported in " + method + " method");
         }
     }
+
+    @Override
+    public V computeIfPresent(K key,
+                              BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(key, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+
+        if (SerializationUtil.isClassStaticAndSerializable(remappingFunction)) {
```

Comment:
IMHO Looks good now. You're going to have to add one more check. We have a feature calling "rolling upgrade" where a cluster can be consisted out of members with 1 minor version difference. This means, if this feature is released in 4.1, the cluster may contain members with version 4.0 which will not be able to deserialise the new entry processor. So, the check should be added:

```
Version clusterVersion = getNodeEngine().getClusterService().getClusterVersion();

        if (SerializationUtil.isClassStaticAndSerializable(remappingFunction)
                && clusterVersion.isGreaterOrEqual(Versions.V4_1)) {
```